pydarn/io/borealis/borealis_convert.py

```diff
@@ -0,0 +1,719 @@
+see: https://borealis.readthedocs.io/en/latest/ 
+
+Future Work
+-----------
+Remove requirement for hdf5 filename by fixing Borealis datawrite so that 
+slice_id is written
+
+"""
+import deepdish as dd
+import h5py
+import logging
+import math
+import numpy as np
+import os
+
+from collections import OrderedDict
+from datetime import datetime
+from typing import Union, List
+
+from pydarn import borealis_exceptions, DarnWrite, borealis_formats
+from pydarn.utils.conversions import dict2dmap
+
+pydarn_log = logging.getLogger('pydarn')
+
+# 3 letter radar code, mapped to station id for DARN files conversion.
+# TODO: when merged with plotting, remove this dictionary and call the 
+#    one in the plotting folder... also move Radars.py to a more 
+#    central location.
+code_to_stid = {
+    "tst": 0,
+    "gbr": 1,
+    "sch": 2,
+    "kap": 3,
+    "hal": 4,
+    "sas": 5,
+    "pgr": 6,
+    "kod": 7,
+    "sto": 8,
+    "pyk": 9,
+    "han": 10,
+    "san": 11,
+    "sys": 12,
+    "sye": 13,
+    "tig": 14,
+    "ker": 15,
+    "ksr": 16,
+    "unw": 18,
+    "zho": 19,
+    "mcm": 20,
+    "fir": 21,
+    "sps": 22,
+    "bpk": 24,
+    "wal": 32,
+    "bks": 33,
+    "hok": 40,
+    "hkw": 41,
+    "inv": 64,
+    "rkn": 65,
+    "cly": 66,
+    "dce": 96,
+    "svb": 128,
+    "fhw": 204,
+    "fhe": 205,
+    "cvw": 206,
+    "cve": 207,
+    "adw": 208,
+    "ade": 209,
+    "azw": 210,
+    "aze": 211,
+    "sve": 501,
+    "svw": 502,
+    "ire": 504,
+    "irw": 505,
+    "kae": 506,
+    "kaw": 507,
+    "eje": 508,
+    "ejw": 509,
+    "she": 510,
+    "shw": 511,
+    "ekb": 512,
+}
+
+
+class BorealisConvert():
+    """
+    Class for converting Borealis filetypes to DARN filetypes.
+
+    See Also
+    --------
+    BorealisRawacf
+    BorealisBfiq
+    Rawacf
+    Iqdat
+    BorealisSiteRead
+    BorealisArrayRead
+    DarnWrite
+
+    Attributes
+    ----------
+    allowed_conversions: dict
+        Mapping of allowed Borealis types to their corresponding 
+        DARN DMap converted types.
+    origin_filetype: str
+    records: OrderedDict{dict}
+        A dictionary of records in the record-by-record format
+    dmap_filename: str
+        The filename of the DARN dmap file to be written.
+    group_names: list[str]
+    dmap_records: list[dict]
+    dmap_filetype: str
+    """
+
+    __allowed_conversions = {'rawacf': 'rawacf', 'bfiq': 'iqdat'}
+
+    def __init__(self, records, origin_filetype, dmap_filename, 
+            hdf5_filename):
+        """
+        Convert HDF5 Borealis records to a given DARN file with DMap format.
+
+        Parameters
+        ----------
+        records: OrderedDict{dict}
+            A dictionary of records in the record-by-record format
+        origin_filetype: str
+            The origin filetype of the Borealis data. 'rawacf' and 'bfiq' 
+            allowed.
+        dmap_filename: str
+            The filename of the DARN dmap file to be written.
+        hdf5_filename: str
```

Comment:
That would be difficult to work into the borealis_formats fields because this slice_id field wouldn't be a required field on read-in but then could be written in the file and then would cause issues when re-read.

Reply:
I'd prefer to leave this as future work for now

Reply:
Sounds good